REST-API-ADVANCED/advanced/src/main/java/com/epam/esm/security/exception/RestAccessDeniedHandler.java

```diff
@@ -0,0 +1,22 @@
+package com.epam.esm.security.exception;
+
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.web.access.AccessDeniedHandler;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Component
+public class RestAccessDeniedHandler implements AccessDeniedHandler {
+    @Override
+    public void handle(HttpServletRequest request, HttpServletResponse response, AccessDeniedException accessDeniedException) throws IOException, ServletException {
+        response.setContentType("application/json");
+        response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+        response.getOutputStream().print("{" +
+                "\"errorCode\":\"" + HttpServletResponse.SC_FORBIDDEN + "\"" +
+                ",\"errorMessage\":\"" + accessDeniedException.getMessage() + "\"}");
+    }
```

Comment:
Can this be done using some kind of json mapper?

Reply:
Yes, ofc. Done